### ISSUE TITLE
"Green Energy Today" / "Grid Import": Trying to understand the values I get from my system

### ISSUE THREAD

**Issue by hzulla** [CJNE/ha-myenergi#394]
Hi there. Forgive me, I'm new to this.

Thanks for the myenergi integration. I'm very thankful for it and am amazed how well it works. Thanks to everybody involved.

This isn't a bug report, but rather a request to explain the values my system reports to me. For the lack of a better forum to do so, I'm asking this here. I didn't find an obvious place to ask this or read up on documentation about the following, but if there is, I'd be glad to get a link to take my question there or to read and learn more about it.

## My home setup

My home has two electricity meters. one meter is used for the house, the second meter is used for the wallbox and the heatpump, using a different contract.

My home does not have solar. All power is consumed from the grid. I will add solar at a later date and bought a "harvi" in preparation for that. It was active for a brief time, but useless without solar. It's still linked to my account, but the "harvi" device is not connected to my installation right now.

![myenergi-products](https://github.com/CJNE/ha-myenergi/assets/1705654/acc122b5-1647-447d-8b76-389977a8ab54)

## Question 1:

Now, using my statistics on myaccount.myenergi.com, have a look at the "Consumed / Exported" energy use statistics for July 31st. It says that my system consumed 0 kWh and imported 0 kWh.

![Screenshot 2023-08-03 at 10-57-58 myenergi account](https://github.com/CJNE/ha-myenergi/assets/1705654/bbd8f99d-3853-4555-9991-1e41e5c6ae73)

Now, check out the "Device use" statistics on myaccount.myenergi.com. It reports a 31.1 kWh charging session on that day.

![Screenshot 2023-08-03 at 10-59-12 myenergi account](https://github.com/CJNE/ha-myenergi/assets/1705654/74148b55-c625-4cd1-9489-b79852989918)

Why does "Consumed / Exported" report 0 kWh imported energy?

## Question 2:

This is what Home Assistant reports for July 31st.

![Screenshot 2023-08-03 at 11-04-34 Verlauf – Home Assistant](https://github.com/CJNE/ha-myenergi/assets/1705654/c3108363-e133-4b1d-9540-5e928bea645a)

It shows the 31.1 kWh charging session. Fine! That's what I want to track.

But also a measurement of "green energy today". I don't have solar, so there's no green energy in my system. What is the roughly 3 kWh of "green energy today" shown there?

## Question 3:

Here's another graph from Home Assistant for July 31st.

![Screenshot 2023-08-03 at 11-09-21 Verlauf – Home Assistant](https://github.com/CJNE/ha-myenergi/assets/1705654/409c4073-9729-4d34-9045-b0fd6a53cf8f)

All the values which are striked through stayed at 0, so I hid them.

I didn't hide "grid import today" on the graph, which also stays at 0 kWh. Why does it stay at 0 kWh? There's no solar. I did import the energy used for charging on that day. Why is it not tracked in "grid import today"?

Why is the "home consumption" at -10W and  "power charging" mirrored at +10W? There is no home consumption of solar on my system.

**Comment by hzulla** [comment]
Is the inactive "harvi" device a possible explanation? Does its existence in my account at myenergi possibly mix up their interpretation of my metering?

**Comment by hzulla** [comment]
Ok, noticed something odd.

I charged the car on two days that week. The "charge added session" ends on a value of 1.27 kWh on July 25th.

When we plugged the car back in on July 31st, the value now doesn't start at 0kWh, but at 3.46 kWh. What's the reason for that?

This oddity also makes the "charge added session" unsuitable to meter the power consumption of the wallbox. Home Assistant thinks the wallbox consumed 33.08 kWh on July 31st, while the myenergi website (see screenshot above) tells me it was 31.1 kWh.

![zappi](https://github.com/CJNE/ha-myenergi/assets/1705654/05009393-1206-46ee-b3aa-d1d05bfa448e)
